File: replacer-backend/src/main/java/es/bvalero/replacer/dump/DumpIndexingStatus.java
package es.bvalero.replacer.dump;

import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Value;
import org.springframework.lang.Nullable;

@Value
@Builder
class DumpIndexingStatus {

    boolean running;

    @Nullable
    Integer numPagesRead;

    @Nullable
    Integer numPagesIndexed;

    @Nullable
    Integer numPagesEstimated;

    @Nullable
    String dumpFileName;

    @Nullable
    LocalDateTime start;

    @Nullable
    LocalDateTime end;

    static DumpIndexingStatus ofEmpty() {
        return DumpIndexingStatus.builder().running(false).build();
    }
}

File: replacer-frontend/src/app/api/models/dump-indexing-status.ts
/* tslint:disable */
/* eslint-disable */

/**
 * Status of the current (or the last) dump indexing
 */
export interface DumpIndexingStatus {

  /**
   * Filename of the indexed dump
   */
  dumpFileName?: string;

  /**
   * Indexing end time (in ms)
   */
  end?: number;

  /**
   * Estimated number of indexable pages
   */
  numPagesEstimated?: number;

  /**
   * Number of indexable pages indexed
   */
  numPagesIndexed?: number;

  /**
   * Number of indexable pages read
   */
  numPagesRead?: number;

  /**
   * If the indexing is running
   */
  running: boolean;

  /**
   * Indexing start time (in ms)
   */
  start?: number;
}
